### ISSUE TITLE
AccordionBox design for a11y

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/sun#359]
It sounds like we'll want to put the expand/collapse button inside the block-level label tag, and potentially other changes.

Area Model will want to pick this up before release (ideally), so I'll be happy to implement once it is designed.

**Comment by terracoda** [comment]
@jonathanolson, I will be creating an official design pattern for this as soon as possible.

I think the HTML will look something like this:
```html
<h3>
  <button aria-expanded="false" aria-controls="collapsible-1">Heading for box 1</button>
</h3>
<div id="collapsible-1" aria-hidden="true">
  <p>Box content 1.</p>
  <p>Content could also be interactive controls like in many sims</p>
</div>

<h3>
  <button aria-expanded="false" aria-controls="collapsible-2">Heading for box 2</button>
</h3>
<div id="collapsible-2" aria-hidden="true">
  <p>Box content 2.</p>
</div>
```
Our A11y API, likely puts parent `div`s around the heading and the container. That's ok :-)

The Javascript will have to manage the aria-expanded and aria-hidden attributes, and the button element is activated with Enter or Space like any button. Focus would remain on the button when toggling between collapsed and expanded states. I think the user should be able to use the down arrow key to read the revealed content in an expanded box on demand.

**Comment by jonathanolson** [comment]
Sounds good.

> I think the user should be able to use the down arrow key to read the revealed content in an expanded box on demand.

I'm not sure how to do that, is that something @jessegreenberg would be familiar with?

Also, I'm interested in what the "ideal" focus highlight should look like (visually) for the button. Is the current behavior good?:

<img width="401" alt="screen shot 2018-05-17 at 9 49 47 am" src="https://user-images.githubusercontent.com/480878/40188657-acc5e278-59b7-11e8-8b5b-4b8e2b12053c.png">

**Comment by terracoda** [comment]
@jonathanolson, an AccordionBox.md file has been created. Please see:
https://github.com/phetsims/sun/blob/master/doc/AccordionBox.md

Does this address your concerns? Can you move forward?

**Comment by terracoda** [comment]
@jonathanolson, I am unassigning myself. Please re-assign if you have further questions.

**Comment by zepumph** [comment]
@terracoda mentioned in the dev meeting today to make sure that the way to get from the expand/collapse button to content is fluid (like with arrow keys or something).

**Comment by zepumph** [comment]
Unassigning until we work on interactive descriptions in a sim with an Accordion box again.

**Comment by zepumph** [comment]
Hey! We are working on a sim with an AccordionBox again (ratio-and-proportion). I think that this issue was all about design, and I see AccordionBox.md is created and it looks complete for the needs of my component. I wonder though if it is still in date, seeing as this issue is really old. 

I think it would be good to confirm with @terracoda. Likely this is just a spot check to tell me that it is alright to use AccordionBox.md as the a11y design for the component. If it is much more than that we may need to re-evaluate.

Tagging @emily-phet since she will likely want to be available of anything related to @terracoda's time.

After @terracoda signs off on the design, we can pick up on https://github.com/phetsims/sun/issues/444.

**Comment by emily-phet** [comment]
Thanks for checking on this @zepumph.
@Matthew-Moore240 and @terracoda let's go over this as part of the description meeting on Tuesday. 

@zepumph - can you share with us (or investigate and share with us) how to access binder? We've all seemed to forgotten...

**Comment by terracoda** [comment]
I started reviewing the markdown file. In general, it is fine. I was taking the opportunity to see what the latest standard had any updates. I haven’t fully digested what I read yet. Also, the example in the ARIA examples is an accordion menu rather than a single expandable/collapsible box, so I wanted to make sure there weren’t any minor nuances.

**Comment by zepumph** [comment]
Binder is updated daily and can be found at https://phetsims.github.io/binder/#other-ui. Accordion Box is right there at the top.

**Comment by zepumph** [comment]
From https://github.com/phetsims/sun/issues/359#issuecomment-653702935 it sounds like there is still work to be done here. I will hold off on any implementation of AccordionBox over in https://github.com/phetsims/sun/issues/444 and (sim -specific) https://github.com/phetsims/ratio-and-proportion/issues/45 until told otherwise in this issue here.

**Comment by emily-phet** [comment]
@zepumph My understanding is that there is a doc (perhaps the markdown file) where @terracoda has reviewed, I think made a few edits, and @Matthew-Moore240 and @terracoda felt good about this for RaP.

@Matthew-Moore240 can you confirm, or clarify if I'm remembering incorrectly? Also, pointing to the doc we looked at in Tuesday's design meeting, to make sure everyone is on the same page, would be helpful.

**Comment by Matthew-Moore240** [comment]
I believe the accordion box in the [RaP design doc](https://docs.google.com/document/d/1HJvvGWYaUv9tlat9S7omQQVng8fGB4FuUhTEM1qx1z8/edit#heading=h.rtza891v0ug4) is good to go. Unless I am missing something here.

**Comment by zepumph** [comment]
Sounds good. I will use what is in the design doc for RaP and the HTML structure/attributes from AccordionBox.md over in https://github.com/phetsims/sun/issues/444. Closing

**Comment by zepumph** [comment]
For completeness, this is what the RaP design doc said about AccordionBox:
```
H3: My Challenge
  Accordion Box
    Accordion Box Title (header): My Challenge
    Accessible Name: H3
      Header contains a title? Yes
      Title is a heading? Yes
      If title is heading, set heading level: H3
      Accordion Title Hidden: No
    Accordion Box Content (panel):
      Parent container accordion box contents: defaults to a div 
      Use same convention as other objects and controls.
      Default open state: expanded
  Help text
    Set left hand and right hand values to explore custom challenge. 

```
